File: selenium-java/src/test/java/frameworks/seleniumjava/pages/LoginPage.java
package frameworks.seleniumjava.pages;

import org.openqa.selenium.WebDriver;
import io.qameta.allure.Step;

public final class LoginPage extends BasePage {
  private static final String USERNAME = "username";
  private static final String PASSWORD = "password";
  private static final String LOGIN_BUTTON = "login-btn";
  private static final String ERROR = "login-error";

  public LoginPage(WebDriver driver) {
    super(driver);
  }

  @Step
  public void login(String user, String pass) {
    typeInto(USERNAME, user);
    typeInto(PASSWORD, pass);
    clickOn(LOGIN_BUTTON);
  }

  @Step
  public String getErrorMessage() {
    return getTestIdText(ERROR);
  }
}

File: selenium-python/pages/login_page.py
from __future__ import annotations

from pages.base_page import BasePage


class LoginPage(BasePage):
    username = "username"
    password = "password"
    login_btn = "login-btn"
    error = "login-error"

    def login(self, username: str, password: str) -> None:
        self.type_into(self.username, username)
        self.type_into(self.password, password)
        self.click_on(self.login_btn)

    def get_error_message(self) -> str:
        return self.get_test_id_text(self.error)
